content/ArtificialIntelligence.txt

```diff
@@ -503,21 +503,54 @@ Problems in training an RNN
 
  - What is a recurrent neural network (RNN)?
 
+What is an attention head?
+
+  An attention head (also described [here](https://towardsdatascience.com/all-you-need-to-know-about-attention-and-transformers-in-depth-understanding-part-1-552f0b41d021#8607)) takes as input a list of embeddings and produces as output a list of attention values.
+
+  An attention head has three matrices of parameters that can be learned like other parameters in a neural network. Let's call these matrices P1, P2, and P3.
+
+  When evaluating an attention head for a specific input, we first make three copies of the input. One copy is called "Queries", one copy is called "Keys", and one copy is called "Values".
+
+  Then, we multiply Queries by P1, Keys by P2, and Values by P3.
+
+  Then, for each token in Queries, we compute the similarity of that query to each other key via a dot product.
+
+  Then, we renormalize the similarities so that the total similarity for each query equals 1.
+
+  Then, for each token in Queries, we multiply each similarity by the corresponding entry from the Values.
+
+  The output of this process is the attention values.
+
+ - What is an embedding?
+ - Neural networks
+ - Training neural networks
+
+What is multi-head attention?
+
+  Multi-head attention is the idea of using and training several attention heads separately, and concatenating their results into a single output.
+
+ - What is an attention head?
+
 What is a transformer?
 
-  Transformers are built on self-attentions -> attend to the most important parts of an input.
+  A [transformer](https://en.wikipedia.org/wiki/Transformer_(deep_learning_architecture)) is a machine learning model that consists of several layers.
+  Each layer contains a multi-headed attention block, and a neural network.
 
-  A transformer (head) has query, key, and value.
-    Query is the current word in a sentence.
-    Keys are all words in a sentence.
-      We compute the similarity between the query and a key, e.g. distance between two encoding vectors.
-      This is the attention weighting: how similar is the key to the query?
-    A value contain actual information for a key.
-      Multiple the attention weighting and the value to compute output. This extract relevant features based on the attention weighting.
+  The attention blocks help the model to identify important words.
+  The neural network allows the model to learn more complicated patterns.
```

Comment:
and generate output?

Reply:
Added a mention of the output of the neural network being the output of the layer